File: src/main/java/com/syfe/personalfinance/dto/ReportDto.java
package com.syfe.personalfinance.dto;

import lombok.*;

import java.math.BigDecimal;
import java.util.Map;

public class ReportDto {

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class MonthlyReportResponse {
        private int month;
        private int year;
        private Map<String, BigDecimal> totalIncome;
        private Map<String, BigDecimal> totalExpenses;
        private BigDecimal netSavings;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class YearlyReportResponse {
        private int year;
        private Map<String, BigDecimal> totalIncome;
        private Map<String, BigDecimal> totalExpenses;
        private BigDecimal netSavings;
    }
}

File: src/main/java/com/syfe/personalfinance/dto/ReportDto.kt
package com.syfe.personalfinance.dto

import java.math.BigDecimal

class ReportDto {

    data class MonthlyReportResponse(
        val month: Int = 0,
        val year: Int = 0,
        val totalIncome: Map<String, BigDecimal> = emptyMap(),
        val totalExpenses: Map<String, BigDecimal> = emptyMap(),
        val netSavings: BigDecimal = BigDecimal.ZERO
    )

    data class YearlyReportResponse(
        val year: Int = 0,
        val totalIncome: Map<String, BigDecimal> = emptyMap(),
        val totalExpenses: Map<String, BigDecimal> = emptyMap(),
        val netSavings: BigDecimal = BigDecimal.ZERO
    )
}
